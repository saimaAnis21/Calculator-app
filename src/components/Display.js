import PropTypes from 'prop-types';
import './styles.css';

function Display(props) {
  const { result } = props;
  return (
    <div className="display-panel h-100 w-100p d-flex jc-flexend ai-center ">
      <span className="pr-10">{result}</span>
    </div>
  );
}

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};

export default Display;
