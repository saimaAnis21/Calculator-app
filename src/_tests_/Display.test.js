import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Display from '../components/Display';
import '@testing-library/jest-dom';

describe('Display', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Display />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the outcome that is given as a prop', () => {
    const { container } = render(<Display result="20" />);
    const displayTxt = container.querySelector('span');
    expect(displayTxt).toHaveTextContent('20');
  });

  it('renders with default value of 0', () => {
    const { container } = render(<Display />);
    const displayTxt = container.querySelector('div');
    expect(displayTxt).toHaveTextContent('0');
  });
});
