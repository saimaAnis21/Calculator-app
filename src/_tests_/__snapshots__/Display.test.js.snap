// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<div
  className="display-panel h-100 w-100p d-flex jc-flexend ai-center "
>
  <span
    className="pr-10"
  >
    0
  </span>
</div>
`;
